common/numerated/src/interval.rs

```diff
@@ -85,352 +67,251 @@ impl<T: Numerated> NonEmptyInterval<T> {
-            unreachable!("`T: UpperBounded + LowerBounded` impl error: must be T::min_value() ≤ T::max_value()")
-        }).into()
-    }
-}
-
-impl<T: Numerated + LowerBounded + UpperBounded, I: Into<T::B>> From<RangeTo<I>> for Interval<T> {
-    fn from(range: RangeTo<I>) -> Self {
-        let end: T::B = range.end.into();
-        let start = T::min_value();
-        match end.unbound() {
-            BoundValue::Value(end) => {
-                debug_assert!(end >= T::min_value());
-                let end = end.dec_if_gt(T::min_value());
-                Self { start, end }
-            }
-            BoundValue::Upper(end) => Self {
-                start,
-                end: Some(end),
-            },
-        }
+        let (start, end) = (T::min_value(), T::max_value());
+        debug_assert!(start <= end, "`T: UpperBounded + LowerBounded` impl error");
+        Self { start, end }
     }
 }
 
-/// Error, which occurs, when trying to convert `start` > `end` into `Interval`.
+/// Trying to make empty interval.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct IntoIntervalError;
+pub struct EmptyRangeError;
 
-impl<T: Numerated, S: Into<T::B>, E: Into<T::B>> TryFrom<(S, E)> for Interval<T> {
-    type Error = IntoIntervalError;
-
-    fn try_from((start, end): (S, E)) -> Result<Self, Self::Error> {
-        use BoundValue::*;
+/// Trying to make interval from range where start > end.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct IncorrectRangeError;
+
+/// Trying to make interval from range where start > end or empty range.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, derive_more::From)]
+pub enum IncorrectOrEmptyRangeError {
+    /// Trying to make empty interval.
+    EmptyRange(EmptyRangeError),
+    /// Trying to make interval start > end.
+    IncorrectRange(IncorrectRangeError),
+}
 
-        let start: T::B = start.into();
-        let end: T::B = end.into();
+impl<T: Numerated + UpperBounded, I: Into<T::Bound>> TryFrom<RangeFrom<I>> for Interval<T> {
+    type Error = EmptyRangeError;
 
-        match (start.unbound(), end.unbound()) {
-            (Upper(start), Upper(_)) => Ok(Self { start, end: None }),
-            (Upper(_), Value(_)) => Err(IntoIntervalError),
-            (Value(start), Upper(end)) => Ok(Self {
-                start,
-                end: Some(end),
-            }),
-            (Value(start), Value(end)) => {
-                if let Some(end) = end.dec_if_gt(start) {
-                    debug_assert!(start <= end);
-                    Ok(Self {
-                        start,
-                        end: Some(end),
-                    })
-                } else if start == end {
-                    Ok(Self { start, end: None })
-                } else {
-                    Err(IntoIntervalError)
-                }
+    fn try_from(range: RangeFrom<I>) -> Result<Self, Self::Error> {
+        match Into::<T::Bound>::into(range.start).unbound() {
```

Comment:
```suggestion
        match T::Bound::from(range.start).unbound() {
```

Reply:
Cannot be done because T::Bound  does not implement  From<I>